From 01908b8f7104d3271461ba96c2497702b62ecb05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Wed, 9 Feb 2022 21:15:15 +0100
Subject: [PATCH 307/515] arm64: dts: rk3399-pinephone-pro: STK3311A INT pin is
 open drain

There's no on-board pull-up resistor and the pin needs to be pulled up,
because it's open drain.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index 35d770e8305d..d3a1bd6bc128 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -851,6 +851,8 @@ light-sensor@48 {
 		reg = <0x48>;
 		interrupt-parent = <&gpio4>;
 		interrupts = <RK_PD3 IRQ_TYPE_EDGE_FALLING>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&stk3311_int>;
 		vdd-supply = <&vcc_3v0>;
 		leda-supply = <&vcc_3v0>;
 	};
@@ -1228,6 +1230,12 @@ display_pwren1: display-pwren1 {
 		};
 	};
 
+	stk3311 {
+		stk3311_int: stk3311-int {
+			rockchip,pins = <4 RK_PD3 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+
 	sound {
 		vcc1v8_codec_en: vcc1v8-codec-en {
 			rockchip,pins = <3 RK_PA4 RK_FUNC_GPIO &pcfg_pull_down>;
-- 
2.35.3

